cpp/src/quantiles/tdigest/tdigest.cu

```diff
@@ -351,13 +351,14 @@ std::unique_ptr<column> percentile_approx(tdigest_column_view const& input,
   // output is a list column with each row containing percentiles.size() percentile values
   auto offsets = cudf::make_fixed_width_column(
     data_type{type_id::INT32}, input.size() + 1, mask_state::UNALLOCATED, stream, mr);
-  auto row_size_iter = thrust::make_constant_iterator(percentiles.size());
+  bool empty_input   = thrust::reduce(input.size_begin(), input.size_begin() + input.size()) == 0;
```

Comment:
This is not safe. The size vales may have something like `-2, 2, -3, 3, ...` which sum to zero.

Let's check if all the size values are zero:
```suggestion
  bool empty_input   = thrust::count_if(rmm::exec_policy(), []__device__() { size[i] == 0} ) == input.size();
```

Reply:
If the input is guaranteed to have size initialized (i.e., the size values are all valid) then `thrust::reduce` is enough. But don't forget to add `rmm::exec_policy`.